.github/workflows/npm-publish-ui.yml

```diff
@@ -0,0 +1,102 @@
+name: Publish OpenHands UI Package
+
+# * Always run on "main"
+# * Run on PRs that have changes in the "openhands-ui" folder or this workflow
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "openhands-ui/**"
+      - ".github/workflows/npm-publish-ui.yml"
+
+# If triggered by a PR, it will be in the same group. However, each commit on main will be in its own unique group
+concurrency:
+  group: npm-publish-ui
+  cancel-in-progress: false
+
+jobs:
+  check-version:
+    name: Check if version has changed
+    runs-on: blacksmith-4vcpu-ubuntu-2204
+    outputs:
+      should-publish: ${{ steps.version-check.outputs.should-publish }}
+      current-version: ${{ steps.version-check.outputs.current-version }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2 # Need previous commit to compare
+
+      - name: Check if version changed
+        id: version-check
+        run: |
+          # Get current version from package.json
+          CURRENT_VERSION=$(cat openhands-ui/package.json | grep '"version"' | sed 's/.*"version": "\(.*\)".*/\1/')
```

Comment:
We should extract the version with jq rather than sed. Structured data enables structured access :)